### PR TITLE
A document within a resolved dossier is no longer movable

### PR DESCRIPTION
**Pull request by elioschmutz**
Jira: https://4teamwork.atlassian.net/browse/NE-241

Currently the `@actions` endpoint always returns the `move_items`, `move_proposal_items` and `move_item` actions if a dossier is closed.

This PR:
- updates the missing `available_expr` for the `move_items` and `move_proposal_items` folder buttons
- respects the dossier state in the documents `is_movable` method which will be used for the `available_expr` of documents

## Before
Bildschirmfoto 2020-11-12 um 12 58 04
Bildschirmfoto 2020-11-12 um 12 57 58

## After
Bildschirmfoto 2020-11-12 um 12 55 22
Bildschirmfoto 2020-11-12 um 12 55 12

The old UI handled this case directly in the [MoveItemsFormView](https://github.com/4teamwork/opengever.core/blob/master/opengever/dossier/move_items.py#L264). So the action was always visible, but it raised an error if it was not possible. With this PR, the action will also no longer be visible in the old ui. The guard still needs to be there because the view can still be accessed directly by modifying the url manually.

It's still possible to move documents out of closed dossiers through the restapis `@move` enpiont. This issue will be handled in a different story: https://4teamwork.atlassian.net/browse/NE-291?searchSessionId=45322838-a049-4f57-be46-737d26b20c9a&searchObjectId=15240&searchContainerId=10023&searchContentType=issue

## Checklist (Must have)

_Everything has to be done/checked. Checked but not present means the author deemed it unnecessary._

- [x] Changelog entry
- [x] Link to issue (Jira or GitHub) and backlink in issue (Jira)

## Checklist (if applicable)

_Only applicable should be left and checked._

- API change:
  - [ ] Documentation is updated
  - If breaking:
    - [ ] api-change label added
    - [ ] #delivery channel notified about breaking change
    - [ ] Scrum master is informed
- New functionality:
  - [x] for `document` also works for `mail`
  - [ ] for `task` also works for `forwarding`
- Upgrade steps (changes in profile):
  - [ ] Make it deferrable if possible
  - [ ] Execute as much as possible conditionally
- DB-Schema migration
  - [ ] All changes on a model (columns, etc) are included in a DB-schema migration.
  - [ ] Constraint names are shorter than 30 characters (`Oracle`)
- [ ] Change could impact client installations, client policies need to be adapted
- New translations
  - [ ] All msg-strings are unicode
- Change in schema definition:
  - [ ] If `missing_value` is specified, then `default` has to be set to the same value

### CODE REVIEW

File: docs/HISTORY.txt [hunk 1]
```diff
@@ -4,6 +4,7 @@ Changelog
 2020.14.0 (unreleased)
 ----------------------
 
+- A document within a closed dossier is no longer movable [elioschmutz]
```
Comment (njohner) [inline docs/HISTORY.txt:8]: I think:
- Disable action to move document within a closed dossier. 

would be closer to the truth, as the document as such is still movable (over the API notably, according to your description in the PR)

File: opengever/document/base.py [hunk 2]
```diff
@@ -147,8 +147,18 @@ def is_inside_a_task(self):
         parent = aq_parent(aq_inner(self))
         return ITask.providedBy(parent)
 
+    def is_inside_a_closed_dossier(self):
+        dossier = get_main_dossier(self)
```
Comment (njohner) [inline opengever/document/base.py:151]: Hum that seems wrong. A document in a closed subdossier but in an open Main dossier should not be movable...

Comment (njohner) [reply opengever/document/base.py:151]: Please also add a test for that

File: opengever/core/upgrades/20201112123011_add_available_expr_for_move_items_folder_actions/upgrade.py [hunk 3]
```diff
@@ -0,0 +1,9 @@
+from ftw.upgrade import UpgradeStep
+
+
+class AddAvailableExprForMoveItemsFolderActions(UpgradeStep):
```
Comment (njohner) [inline opengever/core/upgrades/20201112123011_add_available_expr_for_move_items_folder_actions/upgrade.py:4]: FolderButtons?

**Review by njohner — COMMENTED** [review]
I think your check for having the action or not is not correct, see my comment.

Apart from that LGTM.

File: opengever/document/base.py (hunk 2 above)
Comment (elioschmutz) [reply opengever/document/base.py:151]: 👍 thats true 👀

I was not aware of the ability to close a subdossier

File: opengever/document/base.py [hunk 4]
```diff
@@ -147,8 +147,18 @@ def is_inside_a_task(self):
         parent = aq_parent(aq_inner(self))
         return ITask.providedBy(parent)
 
+    def is_inside_a_closed_dossier(self):
+        dossier = self.get_parent_dossier()
```
Comment (njohner) [inline opengever/document/base.py:151]: I think for a document in an inbox, `self.get_parent_dossier()` will return `None`, so `IDossierMarker.providedBy(dossier)` will be `False` so `is_inside_a_closed_dossier` will return `False` so the document will be movable 😅 . So I think this is OK now, but I'm a bit scared. Can you add a test for a document inside an inbox?

**Review by njohner — APPROVED** [review]
I think it's ok. Can you just make sure that documents inside an inbox are still movable?

Apart from that, LGTM

File: opengever/document/base.py (hunk 4 above)
Comment (elioschmutz) [reply opengever/document/base.py:151]: 🙈 oh man! I'll add such a test 👍

Comment (njohner) [reply opengever/document/base.py:151]: Merci!

**Pull request merged**